features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/ConfigManager.js

```diff
@@ -16,22 +16,35 @@
  * under the License.
  */
 
-import {Axios} from 'axios'
-import DefaultConfig from './DefaultConfigs.json'
+import axios from 'axios'
 
 class ConfigManager {
 
     constructor() {
-        this.preBuildConfigs = DefaultConfig;
-        this.runTimeConfigs = {};
-        this.runTimeConfigLocation = ""; // URL to fetch runtime config JSON
     }
 
-    checkRunTimeConfigs() {
+    /**
+     * get configurations from server: deployment.yaml
+     * @returns {Object}: configuration object
+     */
+    static getConfigs() {
+        return {
+            'promised_environments': function () {
+                if(ConfigManager._promised_environments){
+                    return ConfigManager._promised_environments;
+                }
+                let host = window.location.protocol + "//" + window.location.host;
+                let environmentConfigPath = "/configService/environments";
```

Comment:
Shall we make this a constant and define as a class variable?

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/4692/commits/4510febf6e727f0da1c539e0226e2f1cfb5d283d